database/CoreDatabase.java

```diff
@@ -582,25 +587,45 @@ long committedEventCount() {
         }
     }
 
-    public class StatisticsCorrector {
+    public static class StatisticsCorrector {
 
+        private final CoreDatabase database;
         protected final ConcurrentSet<CompletableFuture<Void>> corrections;
         private final ConcurrentSet<Long> deletedTxnIDs;
-        protected final AtomicBoolean correctionRequired;
+        protected final AtomicReference<State> state;
         protected CoreSession.Data session;
 
-        protected StatisticsCorrector() {
+        protected enum State {INACTIVE, INIT, LOAD, WAITING, CORRECTION_QUEUED, CLOSED}
```

Comment:
we move between various states:
`INACTIVE` - constructed, not ready to set up
`INIT` - we are initialising, but the init hasn't happened yet
`LOAD` - we are loading (init + more work), but the load hasn't happened yet
`WAITING` - the corrector is ready to accept correction jobs
`CORRECTION_QUEUED` - there is a correction job queued
`CLOSED` - the corrector is closed and cannot be opened